rocketpy/motors/tank.py

```diff
@@ -1292,7 +1292,20 @@ def fluid_volume(self):
         Function
             Volume of the fluid as a function of time.
         """
-        return self.liquid_volume + self.gas_volume
+        fluid_volume = self.liquid_volume + self.gas_volume
+
+        # Check if within bounds
+        diff = fluid_volume - self.geometry.total_volume
+
+        if (diff > 1e-6).any():
+            raise ValueError(
+                "The `fluid_volume`, defined as the sum of `gas_volume` and "
+                + "`liquid_volume`, is not equal to the total volume of the tank."
+                + "\n\t\tThe difference is more than 1e-6 m^3 at "
+                + f"{diff.x_array[np.argmin(diff.y_array)]} s."
```

Comment:
I think this error message could be improved a bit

This is a MassBasedTank, meaning the user will have given mass curves to the tank. If this error pops up warning about volume issues, the user might get confused and the correction to this error will not be clear.

I would suggest telling the user that the masses, the densities or the tank geometry might be wrong